include/moqt/publish_track_handler.h

```diff
@@ -0,0 +1,267 @@
+         *
+         *   **Restrictions:**
+         *   - This method cannot be called twice with the same object header group and object IDs.
+         *   - In TrackMode::kStreamPerGroup, ObjectHeaders::group_id **MUST** be different than the previous
+         *     when calling this method when the previous has not been completed yet using PublishContinuationData().
+         *     If group id is not different, the PublishStatus::kPreviousObjectNotCompleteMustStartNewGroup will be returned
+         *     and the object will not be sent. If new group ID is provided, then the previous object will terminate
+         *     with stream closure, resulting in the previous being truncated.
+         *   - In TrackMode::kStreamPerTrack, this method **CANNOT** be called until the previous object has been
+         *     completed using PublishContinuationData(). Calling this method before completing the previous
+         *     object remaining data will result in PublishStatus::kObjectDataIncomplete. No data would be sent and the
+         *     stream would remain unchanged. It is expected that the caller would send the remaining continuation data.
+         *
+         * @note If data is less than ObjectHeaders::payload_length, then PublishObject()
+         *   should be called to send the remaining data.
+         *
+         * @param object_headers        Object headers, must include group and object Ids
+         * @param data                  Payload data for the object, must be <= object_headers.payload_length
+         *
+         * @returns Publish status of the publish
+         *    * PublishStatus::kObjectContinuationDataNeeded if the object payload data is not completed but it was sent,
+         *    * PublishStatus::kObjectDataComplete if the object data was sent and the data is complete,
+         *    * other PublishStatus
+         */
+        PublishStatus PublishObject(const ObjectHeaders& object_headers, BytesSpan data);
+
+        /**
+         * @brief Publish continuation data
+         *
+         * @details Publish continuation data for the previous PublishObject(). The data should be the
+         *      remaining data of the object payload size.
+         *
+         * @param data                  Remaining payload data for the object being published. The length cannot exceed
+         *                              the original object payload size.
+         *
+         * @returns
+         *      * PublishStatus::kObjectDataComplete if the data fulfills the remaining data,
+         *      * PublishStatus::kObjectContinuationDataNeeded if more data is still needed,
+         *      * PublishStatus::kObjectDataTooLarge if the data provided would exceed the remaining amount of data,
+         *      * PublishStatus::kNoPreviousObject if the previous object was completed or if there was no previous object,
+         *      * other PublishStatus
+         */
+        PublishStatus PublishContinuationData(BytesSpan data);
```

Comment:
if the idea is to provide a streaming API , may be something like the below is less confusing for the application developer

 void publish(
      const ObjectHeaders& objHeader,
      uint64_t payloadOffset,
      std::vector<uint8_t>&& payload,
      bool end_of_message);

Reply:
Why would you provide object headers repeatedly for continuation data?  End is also known by end of payload length requirement per https://github.com/moq-wg/moq-transport/blob/main/draft-ietf-moq-transport.md#object-stream.  

A topic for next week is that MoQT doesn't make it easy for object pipelining of unknown length data.  MoQT requires (https://github.com/moq-wg/moq-transport/blob/main/draft-ietf-moq-transport.md#object-stream) `payload_length` before you can even start the transmission of data bytes.  In draft-04 and draft-05 the payload length was not required for datagram or per-object streams, but that has changed in the recent updates (e.g., draft-06) that now requires it for all object messages, even datagram.  This basically results in the API having to require payload length before you can start to transmit.   Hence, we know the end based on the payload length.